lib/admin_credentials.rb

```diff
@@ -43,22 +61,32 @@ def get
         # config_str = res[:stdout].gsub(/^(\s*server:)\s.*$/) {
         #   $1 + " " + env.api_endpoint_url
         # }
-        conf = YAML.load(res[:response])
-        uhash = conf["users"].first # minify should show us only current user
-
-        crt = uhash["user"]["client-certificate-data"]
-        key = uhash["user"]["client-key-data"]
-
-        return APIAccessor.new(
-          id: "admin",
-          client_cert: OpenSSL::X509::Certificate.new(Base64.decode64(crt)),
-          client_key: OpenSSL::PKey::RSA.new(Base64.decode64(key)),
-          env: env
-        )
+        return accessor_from_kubeconfig(res[:response])
       else
         logger.error(res[:response])
         raise "error getting kubeconfig from master #{master_host.hostname}, see log"
       end
     end
   end
+
+  class URLKubeconfigCredentials < AdminCredentials
+    def get
+      url = opts[:spec]
+      res = Http.get(url: url, raise_on_error: true)
+      return accessor_from_kubeconfig(res[:response])
+    end
+  end
+
+  class AutoKubeconfigCredentials < AdminCredentials
+    def get
+      case opts[:spec]
+      when nil
+        MasterOsAdminCredentials.new(env, **opts).get
+      when %r{://}
+        URLKubeconfigCredentials.new(env, **opts).get
+      else
+        raise "unknown gredentials specification: #{opts[:spec]}"
```

Comment:
minor typo should be `credentials` I think...

Reply:
typo from Dvorak keyboard layout?

Reply:
Not really, more likely just being tired when typing that.

Reply:
I make this guess as in Dvorak keyboard layout, "g" is the neighbor of "c". And thanks for make this happen for admin tag cases